sourcegraph/cody
# Autocomplete: add `x-cody-resolved-model` to completion events

Since we have the streaming cancellation propagation issue, we can not rely on the Cody Gateway per model latencies to understand the E2E performance. To clarify the exact model used for every completion, we can use the `x-cody-resolved-model` header and add this information to client-side logs.

- Adds `x-cody-resolved-model` as `resolvedModel` to autocomplete analytics events.
- Closes https://linear.app/sourcegraph/issue/CODY-2321/[autocomplete-latency]-add-cody-gateway-resolved-model-to-client-logs

## Test plan

Updated unit tests. To verify manually:
- Trigger a completion
- Open debug channel
- Notice the new field `gatewayModel` being logged along with the `CompletionLogger:onComplete` event.
